azure/scope/machinepool.go

```diff
@@ -259,6 +262,24 @@ func (m *MachinePoolScope) SetFailureReason(v capierrors.MachineStatusError) {
 	m.AzureMachinePool.Status.FailureReason = &v
 }
 
+func (m *MachinePoolScope) SetBootstrapConditions(provisioningState string, extensionName string) error {
+	switch infrav1.ProvisioningState(provisioningState) {
+	case infrav1.Succeeded:
+		m.V(4).Info("extension provisioning state is succeeded", "vm extension", extensionName, "scale set", m.Name())
+		conditions.MarkTrue(m.AzureMachinePool, infrav1.BootstrapSucceededCondition)
```

Comment:
it'd be nice to set conditions per VMSS VM, hopefully that gets easier with #819

Reply:
Right now it's at the AzureMachinePool.Status level:
```
status:
    conditions:
    - lastTransitionTime: "2021-03-16T23:53:39Z"
      status: "True"
      type: BoostrapSucceeded
    - lastTransitionTime: "2021-03-16T23:49:37Z"
      status: "True"
      type: ScaleSetRunning
```